src/DynamicData/Internal/Rxx.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) 2011-2023 Roland Pheasant. All rights reserved.
+// Roland Pheasant licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for full license information.
+
+#if NET9_0_OR_GREATER
+
+namespace System.Reactive.Linq;
+
+internal static class Rxx
+{
+    public static IObservable<T> Synchronize<T>(this IObservable<T> source, Lock locker) =>
+        new Synchronize<T>(source, locker).Run();
+}
+
+/// <summary>
+/// Keep this class internal as it should be supplied by System.Reactive and probably will be one day.
+/// </summary>
+internal sealed class Synchronize<T>
```

Comment:
I don't see much point in pulling the operator guts out into a separate class here. It was my understanding that we were going to move away from this pattern for our own operators, in favor of doing `partial` classes for the extension fixtures, so we can just split each operator into its own file.

With this just being an attempt to extend RX, that'll likely be temporary, it's also not a big deal, either way.

Reply:
Agreed, and changed.